### PR TITLE
integration tests

### PR DESCRIPTION
**Pull request by justefg**

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -31,23 +85,27 @@ func MakeTestConfig() *AppConfig {
...
 func TestMain(m *testing.M) {
 	InitLogger("debug")
 	events := make(chan *broker.EventMessage)
-	listener := broker.NewEventListener(":8888", events)
-	f := &bytes.Buffer{}
-	a = NewApp(eos.New("https://api.daobet.org"), listener, events, f, MakeTestConfig())
+	listener := new(EventListenerMock)
+	f := &SafeBuffer{}
+	appCfg, keyBag := MakeTestConfig()
+	bc := eos.New("https://api.daobet.org")
```
Comment (algys) [inline main_test.go:106]: after that it's not unit test anymore :)
Don't use external resources in unit tests. For that we will use integration tests.

File: app.go [hunk 2]
```diff
@@ -117,7 +121,7 @@ func (app *App) RunEventProcessor(ctx context.Context, wg *sync.WaitGroup) {
 			for _, event := range eventMessage.Events {
 				go app.processEvent(event)
 			}
-			offset := eventMessage.Events[len(eventMessage.Events)-1].Offset + 1
+			offset := eventMessage.Offset + 1
```
Comment (algys) [inline app.go:124]: why ?

Comment (justefg) [reply app.go:124]: eventMessage.Offset is an offset of the last message in the pack

**Comment by justefg** [comment]
later

**Pull request closed without merging**